feature-libs/customer-ticketing/occ/adapters/occ-customer-ticketing.adapter.ts

```diff
@@ -122,4 +125,80 @@ export class OccCustomerTicketingAdapter implements CustomerTicketingAdapter {
       },
     });
   }
+
+  uploadAttachment(
+    customerId: string,
+    ticketId: string,
+    eventCode: string,
+    file: File
+  ): Observable<unknown> {
+    file = this.converter.convert(file, CUSTOMER_TICKETING_FILE_NORMALIZER);
+    let formData: FormData = new FormData();
+    formData.append('ticketEventAttachment', file);
+
+    return this.http
+      .post(
+        this.getUploadAttachmentEndpoint(customerId, ticketId, eventCode),
+        formData
+      )
+      .pipe(
+        catchError((error) => throwError(normalizeHttpError(error))),
+        this.converter.pipeable(CUSTOMER_TICKETING_FILE_NORMALIZER)
+      );
+  }
+
+  protected getUploadAttachmentEndpoint(
+    customerId: string,
+    ticketId: string,
+    eventCode: string
+  ): string {
+    return this.occEndpoints.buildUrl('uploadAttachment', {
+      urlParams: {
+        customerId,
+        ticketId,
+        eventCode,
+      },
+    });
+  }
+
+  downloadAttachment(
+    customerId: string,
+    ticketId: string,
+    eventCode: string,
+    attachmentId: string
+  ): Observable<unknown> {
+    const httpOptions = {
+      responseType: 'blob' as 'json',
+    };
+    return this.http
+      .get(
+        this.getDownloadAttachmentEndpoint(
+          customerId,
+          ticketId,
+          eventCode,
+          attachmentId
+        ),
+        httpOptions
+      )
+      .pipe(
+        catchError((error) => throwError(normalizeHttpError(error))),
+        this.converter.pipeable(CUSTOMER_TICKETING_FILE_NORMALIZER)
+      );
+  }
```

Comment:
I don't get why you are explicitly doing the download, it should be the browser doing it.